### ISSUE TITLE
Orca doesn't read filenames correctly when I use ctrl+tab

### ISSUE THREAD

**Issue by jvesouza** [microsoft/vscode#140861]
Issue Type: <b>Bug</b>

CC: @isidorn

Steps to reproduce.
1. Make sure that orca is running.
1. Launch vscode and open at least 3 files.
1. Hold down the ctrl key and press the tab key several times.

Expected Result:
For each time the tab key was pressed, orca should announce the name of the file to focus on.

Received Result:
Orca only reads the filename on the first press of the tab key.

VS Code version: Code - Insiders 1.64.0-insider (6612ae0f8b5d9c0d86518ebbd153c724a7cbcf73, 2022-01-17T05:16:08.768Z)
OS version: Linux x64 5.16.0-arch1-1
Restricted Mode: No
Remote OS version: Linux x64 4.14.146-93.123.amzn1.x86_64

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-5200U CPU @ 2.20GHz (4 x 2494)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|1, 2, 2|
|Memory (System)|15.54GB (0.74GB free)|
|Process Argv|--crash-reporter-id bcecf5e0-97d1-4830-b28c-0943c4adab1c --crash-reporter-id bcecf5e0-97d1-4830-b28c-0943c4adab1c|
|Screen Reader|yes|
|VM|0%|
|DESKTOP_SESSION|gnome-xorg|
|XDG_CURRENT_DESKTOP|GNOME|
|XDG_SESSION_DESKTOP|gnome-xorg|
|XDG_SESSION_TYPE|x11|

|Item|Value|
|---|---|
|Remote|SSH: informal|
|OS|Linux x64 4.14.146-93.123.amzn1.x86_64|
|CPUs|Intel(R) Xeon(R) Platinum 8175M CPU @ 2.50GHz (2 x 2500)|
|Memory (System)|3.80GB (0.25GB free)|
|VM|0%|
</details><details><summary>Extensions (35)</summary>

Extension|Author (truncated)|Version
---|---|---
prettier-vscode|esb|9.1.0
shell-format|fox|7.2.2
rest-client|hum|0.24.6
vscode-docker|ms-|1.18.0
csharp|ms-|1.23.17
vscode-aks-tools|ms-|0.0.11
vscode-kubernetes-tools|ms-|1.3.6
python|ms-|2022.0.1612153341-dev
vscode-pylance|ms-|2022.1.1
remote-containers|ms-|0.211.0
remote-ssh|ms-|0.70.0
remote-ssh-edit|ms-|0.70.0
azure-account|ms-|0.9.11
powershell|ms-|2021.12.0
vsliveshare|ms-|1.0.5242
vsliveshare-audio|ms-|0.1.91
oracledevtools|Ora|21.3.0
vscode-boot-dev-pack|Piv|0.1.0
vscode-concourse|Piv|1.29.0
vscode-manifest-yaml|Piv|1.29.0
vscode-spring-boot|Piv|1.29.0
fabric8-analytics|red|0.3.5
java|red|1.2.0
vscode-commons|red|0.0.6
vscode-xml|red|0.18.2
vscode-yaml|red|1.3.0
sonarlint-vscode|Son|3.1.0
vscodeintellicode|Vis|1.2.15
vscode-java-debug|vsc|0.37.0
vscode-java-dependency|vsc|0.18.9
vscode-java-pack|vsc|0.20.0
vscode-java-test|vsc|0.33.1
vscode-maven|vsc|0.34.2
vscode-spring-boot-dashboard|vsc|0.2.0
vscode-spring-initializr|vsc|0.8.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30422396
pythontb:30258533
pythonvspyt551cf:30291413
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
vscod805cf:30301675
pythonvspyt200:30323110
vsccppwt:30421239
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30404738
wslgetstartedc:30421358
vscop804:30404766
vscop453:30404998
vsclayoutctrc:30405799
azactmsal:30409287
vsccppwtstudy:30421978

```

</details>

**Comment by isidorn** [comment]
@jvesouza I just tried this and works just fine for me with VoiceOver and NVDA.
I believe we had some similar discussion in the past, right?

@joanmarie could you please give it a try when you find a chance?

**Comment by jvesouza** [comment]
@isidorn 
Yes, we had some similar discussion in the past, but this problem started recently.
It is not present in version 1.63.

**Comment by isidorn** [comment]
That is strange to hear since we did not update Electron versions in the meantime.
@TylerLeonhardt was there some change in the QuickPick that might have an effect here?

**Comment by joanmarie** [comment]
@jvesouza as a quick check: This isn't an Orca regression, right?

**Comment by jvesouza** [comment]
@joanmarie 
The problem also happens with version 41.1 of orca.

**Comment by joanmarie** [comment]
Ok, I was able to reproduce it locally.
* Works as expected in: 1.64.0-insider Commit: 753319a08 (12 Jan)
* Broken in: 1.64.0-insider Commit: 6612ae0f8 (17 Jan)

I'm afraid I don't have time to bisect. I just hadn't updated my packages in a while and lucked out. :smile: 

The bug is that we're no longer getting focus events every time the selected item changes. When the bug happens, we DO get the initial focus event when the list first pops up.

HTH.

**Comment by isidorn** [comment]
@joanmarie thanks, this helps a lot
So this is a regression from this milestone, let's assign to January.

@TylerLeonhardt if you do not have a clue what might have caused this in the 5 day time frame that @joanmarie identified let me know and I can try to bisect to find the exact commit.

**Comment by TylerLeonhardt** [comment]
Yes, that would be really helpful @isidorn if you don't mind doing that

**Comment by isidorn** [comment]
I'll do it tomorrow. I need to setup orca again on my Linux VM, I already sent a spam email to @joanmarie :)

**Comment by jvesouza** [comment]
@isidorn 
I've never generated the VSCode on my machine and honestly I don't have the slightest idea how to do it, but if the process is not complicated and if someone can send me instructions on how to generate it, I think I can try to find the commit that generated this problem.
Thank you very much.

**Comment by isidorn** [comment]
@jvesouza the process is pretty straight forward. Here are the docs https://github.com/microsoft/vscode/wiki/How-to-Contribute#prerequisites
Let me know how it goes and if there are any questions. I also plan to do some bisection later today.

**Comment by jvesouza** [comment]
@isidorn I'm certainly doing something wrong because I can't find a commit where the problem doesn't happen.
I tried going back to commit 7cb71e89b256d0c9525c24e72c1fb679b4b54004 from November 15th and the problem still appears.
I'm probably regenerating the VSCode the wrong way.

To generate the VSCode I just run the command:
bash scripts/code.sh

**Comment by isidorn** [comment]
@jvesouza you just need to make sure that the `yarn watch` task is running in the background. If it is running, once it builds, you do not even need to start `scripts/code.sh`. You can just reload from your Code OSS running instace.

**Comment by jvesouza** [comment]
@isidorn Unfortunately I still couldn't, although in theory I followed the instructions correctly.

I performed the following steps:
1. yarn watch
2. I waited some time for the compilation process to finish and as the application did not start, I executed the command bash scripts/code-cli.sh in another terminal.
3. I did a test to confirm that the problem was present in this version.
4. In a terminal I ran the command git switch fa553c3ed01b14482da4902ecc12751f2812e7a4
5. I waited some time for the build process to finish and reloaded the application by pressing ctrl+r.
6. I repeated the test again but the problem was there.
7. I tried with other commits but always with the same result.
I'm certainly wrong in some part of the process, but to be honest I don't know where.
Thanks a lot for the help.

**Comment by isidorn** [comment]
Sorry for the slow response. I finally setup my Linux VM so Orca now works nicely for me - great!
@jvesouza you were doing the correct things, but it seems like a `yarn && yarn watch` was needed in between bisect steps. Now I did that and this is the commit that caused the issue
https://github.com/microsoft/vscode/commit/1356392d15bd5434b0b1422483790e9d77010e29

I have just reverted that commit for now. I have verified that it indeed fixes the issue. @jvesouza sorry about the breakage and can you also double check it works for you

@TylerLeonhardt just fyi for the future - the list does not have selection, only focus so we would wrongly set aria-selected to false on focused element.

**Comment by jvesouza** [comment]
@isidorn  I tested it here and it worked perfectly, thanks.

**Comment by isidorn** [comment]
@jvesouza thanks! Adding verified label.
